config/management/src/json_rpc.rs

```diff
@@ -2,46 +2,75 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::error::Error;
+use crate::TransactionContext;
 use libra_secure_json_rpc::JsonRpcClient;
-use std::{fs, path::PathBuf};
-use structopt::StructOpt;
-
-#[derive(Debug, StructOpt)]
-pub struct SubmitTransaction {
-    #[structopt(long)]
-    pub host: String,
-    #[structopt(long)]
-    pub transaction_path: PathBuf,
+use libra_types::account_config::AccountResource;
+use libra_types::validator_config::ValidatorConfig;
+use libra_types::{
+    account_address::AccountAddress, account_state::AccountState, transaction::SignedTransaction,
+};
+
+/// A wrapper around JSON RPC for error handling
+pub struct JsonRpcClientWrapper {
+    client: JsonRpcClient,
 }
 
-impl SubmitTransaction {
-    pub fn execute(self) -> Result<(), Error> {
-        let transaction_path = self.transaction_path.to_str().unwrap().to_string();
-        let data = fs::read(&self.transaction_path)
-            .map_err(|e| Error::UnableToReadFile(transaction_path.clone(), e.to_string()))?;
-        let transaction = lcs::from_bytes(&data)
-            .map_err(|e| Error::UnableToParseFile(transaction_path, e.to_string()))?;
-
-        let client = JsonRpcClient::new(self.host);
-        client.submit_transaction(transaction).map_err(|e| {
-            Error::UnexpectedError(format!("Unable to submit transaction: {}", e.to_string()))
-        })
+impl JsonRpcClientWrapper {
+    pub fn new(host: String) -> JsonRpcClientWrapper {
+        JsonRpcClientWrapper {
+            client: JsonRpcClient::new(host),
+        }
+    }
+
+    pub fn submit_transaction(
+        &self,
+        transaction: SignedTransaction,
+    ) -> Result<TransactionContext, Error> {
+        self.client
+            .submit_transaction(transaction.clone())
+            .map_err(|e| Error::JsonRpcWriteError("transaction", e.to_string()))?;
+        Ok(TransactionContext::new(
+            transaction.sender(),
+            transaction.sequence_number(),
+        ))
+    }
+
+    pub fn get_account_state(&self, account: AccountAddress) -> Result<AccountState, Error> {
```

Comment:
one could argue the get_ here is an anti-pattern

Reply:
Ah, yes that one wasn't cleaned up.  Let's see if I can clean that up.  I'll have to do some commit-fu again.